### DISCUSSION TITLE
mineflayer bot stop lookAt

### DISCUSSION THREAD

**Discussion by MrFiend179** [PrismarineJS/mineflayer#2810] [Q&A]
Ok so Im making a botting app and im using mineflayer library to make the bot, so the issue is I cant make the bot stop looking at player on the chat
``bot.on('chat', function(username, message) {
      if(message === 'look'){
        setInterval(()=>{
          const entity = bot.nearestEntity()
          if (entity !== null) {
            if (entity.type === 'player') {
              bot.lookAt(entity.position.offset(0, 1.6, 0))
            }
          }
        },50)
      }
      else if(message === 'stop'){
        bot.lookAt();
        console.log('stopping')
      }``
This is my code idk what am i doing wrong i tried setting the interval to 0 sec on stop message but that also did nothing

**Comment by rom1504** [comment]
Check setInterval doc, remove interval function

**Reply by MrFiend179** [reply]
I looked at setInterval doc and found clearinterval() might help so I tried it but it doesnt do anything `const nint = setInterval(looker, 50); 
    function looker(){
      if(message === 'look'){
          const entity = bot.nearestEntity()
            if (entity.type === 'player') {
              bot.lookAt(entity.position.offset(0, 1.6, 0))
            }
      }
    }
    if(message === 'stop'){
      clearInterval(nint) 
    } `
This is what I got out of example and as you see the clearinterval was supposed to stop the setinterval but it doesnt do anything the bot still looks at me

**Comment by MrFiend179** [comment]
@rom1504 I tried it without `setinterval()` but then the bot looks at me and doesnt follow me, I need it so that on toggle switch at enable the bot looks at me and on toggle switch disable the bot stops looking at me But I cant seem to find any solution fot it on the internet :/
